core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -641,6 +643,28 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def sendChangeNotifications(workspaceName: WorkspaceName) = {
+
+    def getUserEmails = {
+      dataSource.inTransaction{ dataAccess =>
+        withWorkspaceContext(workspaceName, dataAccess) {workspaceContext =>
+          requireAccessIgnoreLock(workspaceContext.workspace, WorkspaceAccessLevels.Write, dataAccess) {
+            DBIO.sequence(workspaceContext.workspace.accessLevels.values.map { group =>
+              dataAccess.rawlsGroupQuery.flattenGroupMembership(group)
+            }.map(_.map(user => userInfo.userEmail)))
+          }
+        }
+      }
+    }
+    for {
+      emails <- getUserEmails
+    } yield {
+      val notificationMessages = emails.map(email => Notifications.WorkspaceChangedNotification(email, workspaceName))
+      notificationDAO.fireAndForgetNotifications(notificationMessages)
+    }
```

Comment:
yeah - I think I was just so excited by finally grasping that syntax I had to use it... but cleaner in this case not to, so making the change...